docs/design/mp_arktos.md

```diff
@@ -0,0 +1,229 @@
+
+Please note that any of the three fields can be missing. For example, user might only specify  ```ip``` field; in this case, Mizar needs to figure out which subnet this ip falls into, and creates the pod into the correct subnet. 
+
+## Network Object Association
+
+In Mizar, the association of all network objects are stored as key/value sets in Mizar’s object store. For instance, 
+
+The Arktos network object is associated with Mizar vpc as: 
+	```self.store.vpcs_arktosnet_store[arktosnet.name] = {}```
+    ```self.store.vpcs_arktosnet_store[arktosnet.name][vpc.name] = vpc_object```
+
+The Arktos network object is associated with services in similar ways: 
+    ```self.store.services_arktosnet_store[arktosnet.name] = {}```
+    ```self.store.services_arktosnet_store[arktosnet.name][service.name] = service_object```
+
+In addition, Mizar already provides key/value sets to store relationships between endpoints/pods, subnets/nets and vpcs: 
+
+* ```nets_vpc_store```
+* ```eps_net_store```
+
+## Arktos Network Object Creation 
+
+For now, Mizar will only support VPC object and Arktos network object association, meaning Arktos must pass in a vpc id that already exist. 
+
+Mizar has operator that is in charge of managing lifecycle of Arktos network object. 
+
+In case of flat network: 
+```yaml
+apiVersion: arktos.futurewei.com/v1
+kind: Network
+metadata:
+  name: default
+spec:
+  type: flat
+```
+Operator kicks off Arktos network object ```default``` creation process, and then associated with the default network ```vpc0``` in Mizar:
+```self.store.vpcs_arktosnet_store[default][vpc0] = vpc0 object```
+
+In case of vpc network: 
+```yaml
+apiVersion: arktos.futurewei.com/v1
+kind: Network
+metadata:
+  name: vpc-1
+spec:
+  type: mizar
+  vpcID: vpc-1a2b3c4d
+```
+Operator starts creation process for Arktos network object ```vpc-1```, and then search for existing vpc object in Mizar to find vpc with id = vpc-1a2b3c4d. Once the vpc object is found, it then associated with the newly created Arktos network object ```vpc-1```: 
+    ```self.store.vpcs_arktosnet_store[vpc-1][vpc-1a2b3c4d] = vpc-1a2b3c4d object```
+
+## Pod Creation
+ Sample Pod definition:
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+  labels:
+    arktos.futurewei.com/network: vpc-1
+  annotations:
+    arktos.futurewei.com/nic: {"name": "eth0", "ip": "10.10.1.12", "subnet": "net1"}
+    arktos.futurewei.com/vip-hint: {"cidr":"10.10.1.0/26", "ip":"10.10.1.12", "policy":"BestEffort"}
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    ports:
+      - containerPort: 443
+```
+Operator kicks off creation process:
+* List Kubernetes cluster object / Arktos newtork object
+* Retrieve ```labels``` and ```annotations``` information: 
+    * ```arktos.futurewei.com/network```: vpc-1
+    * ```arktos.futurewei.com/nic```: {"name": "eth0", "ip": "10.10.1.12", "subnet": "net1"}
+* Find Arktos network object that is listed in ```arktos.futurewei.com/network```
```

Comment:
suggest to use mizar.futurewei.com/network-user-input, as this annotation is mizar specific.